File: src/entities/todo.ts
export class Todo {
  private id: number;
  private title: string;
  private status: string;
  private userId: number;
  private createdAt: Date;
  private finishedAt: Date | null;

  constructor(
    id: number,
    title: string,
    status: string,
    userId: number,
    createdAt: Date,
    finishedAt: Date | null,
  ) {
    this.id = id;
    this.title = title;
    this.status = status;
    this.userId = userId;
    this.createdAt = createdAt;
    this.finishedAt = finishedAt;
  }

  getId(): number {
    return this.id;
  }

  getTitle(): string {
    return this.title;
  }

  getStatus(): string {
    return this.status;
  }

  getUserId(): number {
    return this.userId;
  }

  getCreatedAt(): Date {
    return this.createdAt;
  }

  getFinishedAt(): Date | null {
    return this.finishedAt;
  }

  public updateTitle(newTitle: string): void {
    if (newTitle.length <= 0) {
      throw new Error('title should not empty');
    }
    this.title = newTitle;
  }

  public start(): void {
    this.status = 'doing';
  }

  public done(): void {
    this.status = 'done';
  }
}

File: src/main/java/com/todoapp/entities/Todo.java
package com.todoapp.entities;

import jakarta.persistence.*;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Entity
@Table(name = "todos")
public class Todo {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "title", nullable = false, length = 255)
    private String title;

    @Column(name = "status", nullable = false)
    @Enumerated(EnumType.STRING)
    private Status status;
    
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "finished_at")
    private LocalDateTime finishedAt;
    
    public Todo() {
        this.status = Status.READY;
        this.createdAt = LocalDateTime.now();
    }
    
    public Todo(String title) {
        this();
        this.title = title;
    }
    
    public void updateTitle(String newTitle) {
        this.title = newTitle;
    }
    
    public void start() {
        if (this.status == Status.READY) {
            this.status = Status.DOING;
        }
    }
    
    public void done() {
        this.status = Status.DONE;
    }
}